account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -149,9 +183,85 @@ export async function createSMAV2Account(
     ]) as Promise<bigint>;
   };
 
+  const accountContract = getContract({
+    address: baseAccount.address,
+    abi: modularAccountAbi,
+    client,
+  });
+
+  const getExecutionData = async (args: ExecutionDataParams) => {
+    if (!(await baseAccount.isAccountDeployed())) {
+      return {} as ExecutionDataView;
+    }
+
+    const { selector } = args;
+    return await accountContract.read.getExecutionData([selector]);
+  };
+
+  const getValidationData = async (args: ValidationDataParams) => {
+    if (!(await baseAccount.isAccountDeployed())) {
+      return {} as ValidationDataView;
+    }
+
+    const { validationModuleAddress, entityId } = args;
+    return await accountContract.read.getValidationData([
+      serializeModuleEntity({
```

Comment:
if validation module is provided, entityid should be 0xFFFFFFFF. if entity id is provided, validation module should be 0